app/controllers/application/accessrequest/RequestProductionAccessStartController.scala

```diff
@@ -39,23 +39,30 @@ class RequestProductionAccessStartController @Inject()(
     applicationAuth: ApplicationAuthActionProvider,
     appConfig: FrontendAppConfig,
     applicationApiBuilder: ApplicationApiBuilder,
-    navigator: Navigator
+    navigator: Navigator,
+    hipEnvironments: HipEnvironments,
+    errorResultBuilder: ErrorResultBuilder
 )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
-  def onPageLoad(id: String): Action[AnyContent] = (identify andThen applicationAuth(id)).async {
+  def onPageLoad(id: String, environmentId: String): Action[AnyContent] = (identify andThen applicationAuth(id)).async {
     implicit request =>
-      for {
-        applicationApis <- applicationApiBuilder.build(request.application)
-        userAnswers <- Future.fromTry(
-          UserAnswers(
-            id = request.identifierRequest.user.userId,
-            lastUpdated = clock.instant()
-          )
-          .set(RequestProductionAccessApplicationPage, request.application)
-            .flatMap(_.set(RequestProductionAccessApisPage, applicationApis))
-        )
-        _ <- accessRequestSessionRepository.set(userAnswers)
-      } yield Redirect(navigator.nextPage(RequestProductionAccessStartPage, NormalMode, userAnswers))
+      hipEnvironments.forUrlPathParameter(environmentId) match {
+        case Some(hipEnvironment) if hipEnvironment.isProductionLike =>
+          for {
+            applicationApis <- applicationApiBuilder.build(request.application)
+            userAnswers <- Future.fromTry(
+              UserAnswers(
+                id = request.identifierRequest.user.userId,
+                lastUpdated = clock.instant()
+              )
+                .set(RequestProductionAccessApplicationPage, request.application)
+                .flatMap(_.set(RequestProductionAccessApisPage, applicationApis))
+                .flatMap(_.set(RequestProductionAccessEnvironmentIdPage, hipEnvironment.id))
+            )
+            _ <- accessRequestSessionRepository.set(userAnswers)
+          } yield Redirect(navigator.nextPage(RequestProductionAccessStartPage, NormalMode, userAnswers))
+        case _ => Future.successful(errorResultBuilder.notFound())
```

Comment:
not sure if you prefer this?
```suggestion
        case _ => Future.successful(errorResultBuilder.environmentNotFound(environmentId))
```

Reply:
I think we've decided to stick with generic Not Found messages where users must be URL hopping. The application would never link to a page with an invalid or non-production-like environment.